tests/foreman/ui/test_architecture.py

```diff
@@ -32,14 +34,17 @@ def test_positive_create_arch_with_os(self, test_data):
         @Assert: Architecture is created
 
         """
-        entities.OperatingSystem(name=test_data['os_name']).create()
         with Session(self.browser) as session:
-            make_arch(session, name=test_data['name'],
-                      os_names=[test_data['os_name']])
-            self.assertIsNotNone(self.architecture.search(test_data['name']))
-
-    @data(*generate_strings_list())
-    def test_positive_create_arch_with_different_names(self, name):
+            for test_data in valid_arch_os_names():
+                with self.subTest(test_data):
+                    entities.OperatingSystem(
+                        name=test_data['os_name']).create()
+                    make_arch(session, name=test_data['name'],
```

Comment:
This is one of those bugs which was hidden so far with smoke=1 flag and they are all surface now as we are testing all data points :)

So it turns out that this specific error is on our locator.  We use the following locator to identify the architecture name link:
```sh
//a[contains(., '%s')]
```
When you use non-html characters, it is happy.  When you use html characters, it stops identifying the element because characters like <, /, > are escaped in the DOM object.  @omaciel helped me identify a workaround for this and I will add that in a new commit in the same PR.  We may eventually need that fix for all entities, but I will know more when working on other entities.